src/main/java/com/checkmarx/flow/service/ASTScanner.java

```diff
@@ -35,7 +35,11 @@ protected ScanResults toScanResults(ASTResultsWrapper internalResults) {
 
     @Override
     protected String getScanId(ASTResultsWrapper internalResults) {
-        return internalResults.getAstResults().getScanId();
+        if (internalResults.getAstResults() != null && internalResults.getAstResults().getResults() !=null){
+            return internalResults.getAstResults().getResults().getScanId();
```

Comment:
internalResults.getAstResults() is getting calculated 3 times, moreover, try to use Optional.ofNullable :)